### PR TITLE
Guard+rounting+lazyloading for config/activities (provider/parent)

### PR DESCRIPTION
**Pull request by litvinets**

### CODE REVIEW

File: src/app/app-routing.module.ts [hunk 1]
```diff
@@ -1,6 +1,7 @@
 import { NgModule } from '@angular/core';
 import { Routes, RouterModule } from '@angular/router';
 
+
```
Comment (dmitryy90) [inline src/app/app-routing.module.ts:4]: why?

File: src/app/header/header.component.ts [hunk 2]
```diff
@@ -47,4 +48,10 @@ export class HeaderComponent implements OnInit {
   login(): void{
     this.store.dispatch(new Login());
   }
+  linksGenerating(){
+    let role = this.store.selectSnapshot<string>(UserRegistrationState.role);
+    let urlRole = (role==='organization') ? 'provider':'parent';
```
Comment (dmitryy90) [inline src/app/header/header.component.ts:53]: you can move it to enum and use it as an associative array

File: src/app/shell/parent/parent.guard.ts [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  providedIn: 'root'
+})
+export class ParentGuard implements CanLoad {
+  constructor(public store: Store) {}
+
+  canLoad(
+    route: Route,
+    segments: UrlSegment[]): Observable<boolean | UrlTree> | Promise<boolean | UrlTree> | boolean | UrlTree {
+      const role=this.store.selectSnapshot<string>(UserRegistrationState.role);
+    if(role==='parent'){
```
Comment (dmitryy90) [inline src/app/shell/parent/parent.guard.ts:17]: return role === 'parent';

**Review by dmitryy90 — CHANGES_REQUESTED** [review]

**Review by dmitryy90 — APPROVED** [review]

File: src/app/header/header.component.ts (hunk 2 above)
Comment (litvinets) [reply src/app/header/header.component.ts:53]: I didn't understand how or what to do;
I will do it later

**Pull request merged**